play.pokemonshowdown.com/src/panel-mainmenu.tsx

```diff
@@ -793,23 +796,7 @@ export class TeamForm extends preact.Component<{
 		}
 	};
 	render() {
-		if (window.BattleFormats) {
-			const starredPrefs = PS.prefs.starredformats || {};
-			// .reverse() because the newest starred format should be the default one
-			const starred = Object.keys(starredPrefs).filter(id => starredPrefs[id] === true).reverse();
-			if (!this.format) {
-				this.format = `gen${Dex.gen}randombattle`;
-				for (let id of starred) {
-					let format = window.BattleFormats[id];
-					if (!format) continue;
-					if (this.props.selectType === 'challenge' && format?.challengeShow === false) continue;
-					if (this.props.selectType === 'search' && format?.searchShow === false) continue;
-					if (this.props.selectType === 'teambuilder' && format?.team) continue;
-					this.format = id;
-					break;
-				}
-			}
-		}
+		this.format = toID(PS.prefs.lastselectedformat?.[this.props.selectType!] || `gen${Dex.gen}randombattle`);
```

Comment:
```suggestion
		this.format ||= toID(PS.prefs.lastselectedformat?.[this.props.selectType!] || `gen${Dex.gen}randombattle`);
```

Arguably even this is not enough; we'd need to write a system that can wait for prefs to be loaded.

Reply:
that was not an issue here tho, prefs were also loaded